### ISSUE TITLE
router basePath parameter in config doesn't work with reverse proxy

### ISSUE THREAD

**Issue by Pierric82** [owntracks/frontend#102]
Hi,
Apologies if I'm missing something obvious, but I was trying to set up this frontend as `track.mydomain.com/front`, whereby the recorder server runs on the root of `track.mydomain.com`.
In my nginx reverse proxy I've tried various ways to rewrite the URL so that `track.mydomain.com/front*` is sent to this frontend, and I've correspondingly set up `router.basePath` to be `/front`.
However if the reverse proxy manages to redirect the query for '/front' to the frontend, the javascript and CSS is then loaded from `/css/...`, `/js/...`, not taking into account the base path I've set up. So those files are not found because they are not served by the recorder, who receives the requests instead of the frontend.
If I access directly the frontend by bypassing the reverse proxy, I see that once the core files are loaded, the configuration is applied, as it adds `/front` to the URL after initially loading successfully (then, reloading the URL that the frontend provided with `/front/` in it fails, I suppose it expects a reverse proxy to rewrite the URL back to the version without `/front`; that would make sense).
It seems to me that since the `/index.html` file does NOT rewrite the URLs of its key files, this cannot work. But have I missed something?

**Comment by linusg** [comment]
> It seems to me that since the /index.html file does NOT rewrite the URLs of its key files, this cannot work. But have I missed something?

All resources referenced by `index.html` are relative (see [here](https://github.com/owntracks/frontend/blob/f5389b84abbd4ab203502e7813252181835612ea/vue.config.js#L9) and [here](https://cli.vuejs.org/config/#publicpath) for details), so this should just work if you serve it from `/front`.

I won't be able to help you with a complex nginx setup as I've switched exclusively to traefik years ago, but if you post some config snippets / browse network screenshots we might be able to debug this further :)

**Comment by Pierric82** [comment]
Hi Linus,

Thanks for your supportive response, appreciate it!

You are right, it was indeed me who failed to set up nginx correctly, as the relative paths should not cause a problem indeed! In the end I got it to almost work by having 2 `rewrite` lines in the configuration:
```
location /front {
        rewrite ^/front/(.*)    /$1 break;
        rewrite ^/front    / break;
        proxy_pass      http://$upstream_app:$upstream_port_front/$uri$is_args$args;
        proxy_http_version  1.1;
        proxy_set_header    Host $host;
        proxy_set_header    X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header    X-Real-IP $remote_addr;
}
```

The 2nd line ensures that the initial URL without trailing slash does get the root HTML file correctly, while the first ensures that the browser does not consider that "css" relative to "/front" means "/css" (whereas "css" relative to "/front/" would mean "/front/css"), which I now understand was my main problem before.

Now the only issue left is that everything seems to load and respond properly, including menus, versions, etc., except for the map which is just a white space. When accessing the frontend directly by IP address, bypassing the proxy, I don't have that issue. If you have any thoughts on this, let me know!

Thanks,
Pierric

**Comment by linusg** [comment]
glad you (almost) got it working! The missing map issue came up before - the discussions in #55, #89 might be helpful?

**Comment by Pierric82** [comment]
I'm so sorry, this was just me having left things in an unclean state yesterday while investigating. I had commented out the router.basePath configuration and left it there, so it was now loading "a bit more" (the CSS and JS was loading), but calls were then made to /api instead of /front/api (I had missed that the one call to /api/0/list was failing). In that sense yes this was related to issue 89. Fixed this and now everything is working! 

Thanks again!
